File: 240514/오목/O-mok.java
import java.util.*;
public class Main {
    static boolean check = false;
    static int x = 0;
    static int y = 0;
    static int winnerNum = 0;
    public static boolean checkWin(int[][] plate) {
        //가로로 이겼을 경우
        for(int i=0;i<19;i++) {
            for(int j=0;j<14;j++) {
                if(plate[i][j]==1 && plate[i][j+1]==1 && plate[i][j+2]==1 && plate[i][j+3]==1 && plate[i][j+4]==1){
                    x = i;
                    y = j+2;
                    winnerNum = 1;
                    return true;
                }
                if(plate[i][j]==2 && plate[i][j+1]==2 && plate[i][j+2]==2 && plate[i][j+3]==2 && plate[i][j+4]==2){
                    x = i;
                    y = j+2;
                    winnerNum = 2;
                    return true;
                }
            }
        }
        //세로로 이겼을 경우
        for(int i=0;i<14;i++) {
            for(int j=0;j<19;j++) {
                if(plate[i][j]==1 && plate[i+1][j]==1 && plate[i+2][j]==1 && plate[i+3][j]==1 && plate[i+4][j]==1){
                    x = i+2;
                    y = j;
                    winnerNum = 1;
                    return true;
                }
                if(plate[i][j]==2 && plate[i+1][j+1]==2 && plate[i+2][j]==2 && plate[i+3][j]==2 && plate[i+4][j]==2){
                    x = i+2;
                    y = j;
                    winnerNum = 2;
                    return true;
                }
            }
        }
        //대각선으로 이겼을 경우
        for(int i=0;i<14;i++) {
            for(int j=0;j<14;j++) {

                if(plate[i][j]==1 && plate[i+1][j+1]==1 && plate[i+2][j+2]==1 && plate[i+3][j+3]==1 && plate[i+4][j+4]==1){
                    x = i+2;
                    y = j+2;
                    winnerNum = 1;
                    return true;
                }
                if(plate[i][j]==2 && plate[i+1][j+1]==2 && plate[i+2][j+2]==2 && plate[i+3][j+3]==2 && plate[i+4][j+4]==2){
                    x = i+2;
                    y = j+2;
                    winnerNum = 2;
                    return true;
                }
            }
        }
        for(int i=4;i<19;i++) {
            for(int j=4;j<19;j++) {
                if(plate[i][j]==1 & plate[i-1][j-1]==1 &&plate[i-2][j-2]==1 && plate[i-3][j-3]==1 && plate[i-4][j-4]==1) {
                    x = i-2;
                    y = j-2;
                    winnerNum = 1;
                    return true;
                }
                if(plate[i][j]==1 & plate[i-1][j-1]==2 &&plate[i-2][j-2]==2 && plate[i-3][j-3]==2 && plate[i-4][j-4]==2) {
                    x = i-2;
                    y = j-2;
                    winnerNum = 2;
                    return true;
                }
            }
        }
        return false;
    }

    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int[][] plate = new int[19][19];

        for(int i=0;i<19;i++) {
            for(int j=0;j<19;j++) {
                plate[i][j] = sc.nextInt();
            }
        }

        for(int i=0;i<19;i++) {
            if(check == true) {
                break;
            }
            for(int j=0;j<19;j++) {
                if(checkWin(plate)==true) {
                    check = true;
                    break;
                }
            }
        }
        if(winnerNum ==0) {
            System.out.println(winnerNum);
        }
        else if(winnerNum == 1) {
            System.out.println(winnerNum);
            x++; y++;
            System.out.print(x + " " +y);
        }
        else if(winnerNum == 2) {
            System.out.println(winnerNum);
            x++; y++;
            System.out.print(x+ " " +y);
        } else {
            return;
        }

    }
}

File: 240517/오목/O-mok.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    val omokBoard = Array(19) { IntArray(19) { sc.nextInt() } }
    val dx = intArrayOf(0, 1, -1, -1, 1, 1, 0, 0)
    val dy = intArrayOf(1, -1, 1, -1, 1, -1, 0, -1)

    for (i in 0 until 19) {
        for (j in 0 until 19) {
            if (omokBoard[i][j] == 1) {

                for (k in 0 until 4) {
                    val nx1 = i + dx[k]
                    val ny1 = j + dy[k]
                    val nx2 = i + dx[k] * 2
                    val ny2 = j + dy[k] * 2
                    val nx3 = i + dx[8 - k - 1]
                    val ny3 = j + dy[8 - k - 1]
                    val nx4 = i + dx[8 - k - 1] * 2
                    val ny4 = j + dy[8 - k - 1] * 2

                    if (inRange(nx1, ny1) &&
                        inRange(nx2, ny2) &&
                        inRange(nx3, ny3) &&
                        inRange(nx4, ny4)
                    ) {

                        if (omokBoard[nx1][ny1] == 1 &&
                            omokBoard[nx2][ny2] == 1 &&
                            omokBoard[nx3][ny3] == 1 &&
                            omokBoard[nx4][ny4] == 1
                        ) {
                            println(1)
                            print("${i + 1} ${j + 1}")
                            return
                        }
                    }
                }

            } else if (omokBoard[i][j] == 2) {

                for (k in 0 until 4) {
                    val nx1 = i + dx[k]
                    val ny1 = j + dy[k]
                    val nx2 = i + dx[k] * 2
                    val ny2 = j + dy[k] * 2
                    val nx3 = i + dx[8 - k - 1]
                    val ny3 = j + dy[8 - k - 1]
                    val nx4 = i + dx[8 - k - 1] * 2
                    val ny4 = j + dy[8 - k - 1] * 2

                    if (inRange(nx1, ny1) &&
                        inRange(nx2, ny2) &&
                        inRange(nx3, ny3) &&
                        inRange(nx4, ny4)
                    ) {

                        if (omokBoard[nx1][ny1] == 2 &&
                            omokBoard[nx2][ny2] == 2 &&
                            omokBoard[nx3][ny3] == 2 &&
                            omokBoard[nx4][ny4] == 2
                        ) {
                            println(2)
                            print("${i + 1} ${j + 1}")
                            return
                        }
                    }
                }

            }


        }
    }

    print(0)
}

fun inRange(x: Int, y: Int): Boolean {
    return x in 0 until 19 && y in 0 until 19
}